### PR TITLE
CONFLUENT: Remove support-metrics-client

### PR DESCRIPTION
**Pull request by ijuma**
Removed legacy module to reduce maintenance overhead.

Moved `SupportedKafka` to core module, deprecated it
and simplified it. It now just forwards to `Kafka`.

Fixed some compiler warnings in support-metrics-common.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
cc @cotedm

File: tests/kafkatest/services/kafka/kafka.py [hunk 1]
```diff
@@ -368,7 +368,7 @@ def start_node(self, node, timeout_sec=60):
     def pids(self, node):
         """Return process ids associated with running processes on the given node."""
         try:
-            cmd = "jcmd | grep -e %s -e %s | awk '{print $1}'" % (self.java_class_name(), self.apache_java_class_name())
+            cmd = "jcmd | grep -e %s -e %s | awk '{print $1}'" % self.java_class_name()
```
Comment (cmccabe) [inline tests/kafkatest/services/kafka/kafka.py]: This should be ( self.java_class_name(), self.java_class_name() ) since there are two %s

Comment (ijuma) [reply tests/kafkatest/services/kafka/kafka.py]: Yeah, I meant to remove one of the %s, good catch.

Comment (ijuma) [reply tests/kafkatest/services/kafka/kafka.py]: Fixed.

File: build.gradle [hunk 2]
```diff
@@ -950,8 +948,6 @@ project(':core') {
     from(project(':streams:test-utils').configurations.runtime) { into("libs/") }
     from(project(':support-metrics-common').jar) { into("libs/") }
     from(project(':support-metrics-common').configurations.runtime) { into("libs/") }
```
Comment (ijuma) [inline build.gradle]: @cmccabe I'm thinking we can remove this too. The reasoning is that we don't need this in the broker classpath, but other projects still depend on `support-metrics-common` as a library. What do you think?

Comment (ijuma) [reply build.gradle]: Actually, maybe we don't need to do this since `kafka-run-class` does not include `support-metrics-common` in the classpath.

**Comment by ijuma** [comment]
Downstream failure is unrelated:

> [INFO:2020-02-05 01:16:10,534]: [ERROR] Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:3.0.2:jar (default-jar) on project ksql-docker: You have to use a classifier to attach supplemental artifacts to the project instead of replacing them. -> [Help 1]

One unrelated flaky test failure:

> kafka.admin.DescribeConsumerGroupTest.testDescribeGroupStateWithShortInitializationTimeout

File: bin/kafka-server-start.sh [hunk 3]
```diff
@@ -50,4 +50,4 @@ case $COMMAND in
     ;;
 esac
 
-exec $base_dir/kafka-run-class.sh $EXTRA_ARGS io.confluent.support.metrics.SupportedKafka "$@"
+exec $base_dir/kafka-run-class.sh $EXTRA_ARGS kafka.Kafka "$@"
```
Comment (omkreddy) [inline bin/kafka-server-start.sh]: we should also update `/bin/kafka-server-stop.sh`?

Comment (ijuma) [reply bin/kafka-server-start.sh]: Because we still have the SupportedKafka class (even though deprecated), the stop script has to understand it still.

**Comment by ijuma** [comment]
Good point about the config and muckrake test. Will make the changes.

**Comment by omkreddy** [comment]
we should also remove `support-metrics-bundle` script?

**Comment by ijuma** [comment]
@omkreddy Updated the PR.

**Comment by ijuma** [comment]
@omkreddy https://github.com/confluentinc/muckrake/pull/949

**Comment by ijuma** [comment]
Removed a couple of tests that are no longer necessary and fixed some compiler warnings.

**Comment by ijuma** [comment]
Unrelated flaky test failures and unrelated downstream job failure:

> [INFO:2020-02-07 15:58:23,248]: [ERROR] Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:3.0.2:jar (default-jar) on project ksql-docker: You have to use a classifier to attach supplemental artifacts to the project instead of replacing them. -> [Help 1]

**Pull request merged**